### ISSUE TITLE
[Feature] Support change member cluster kubeconfig 

### ISSUE THREAD

**Issue by zryfish** [kubesphere/kubesphere#4426]
**What's it about?**
Multicluster is widely used since we introduced it from version v3.0. There is one problem with these member clusters,  there is no way to replace expired `kubeconfig`. You can use the following command to check expiration date of member cluster kubeconfig:
```
kubectl get cluster [cluster name] -o jsonpath='{.spec.connection.kubeconfig}' | base64 -d | grep "client-certificate-data" | awk '{print $2}' | base64 -d | openssl x509 -enddate -noout -in -
```
After the expiration date, the user either needs to unbind cluster and rejoin cluster or manually change `kubeconfig` in cluster cr. Both are inconvenient and complicated. 

**What's the reason why we need it?**

Users can replace member cluster kubeconfig through UI wich is less risky.
/area multicluster
/kind feature-request

/cc @kubesphere/sig-multicluster

**Comment by zryfish** [comment]
Please take a look at this @iawia002

**Comment by zryfish** [comment]
We may also need to find a way to notify users when cluster kubeconfig approaching the expiration date.

**Comment by zryfish** [comment]
@iawia002  Any update on this?

**Comment by iawia002** [comment]
> @iawia002 Any update on this?

I will submit a PR in a few days to add the corresponding API

**Comment by iawia002** [comment]
> We may also need to find a way to notify users when cluster kubeconfig approaching the expiration date.

If we want to achieve this, we may need to add the openssl packages to the base image of ks-apiserver. Is this acceptable?

**Comment by zryfish** [comment]
Just go ahead, we can review your solution once it's more concrete.

**Comment by zryfish** [comment]
/kind design
/area console

@iawia002 Could you add some documentation about the new API?

**Comment by iawia002** [comment]
> @iawia002 Could you add some documentation about the new API?

Yes, where should I put this documentation?

**Comment by zryfish** [comment]
Need to validate the old cluster with the new kubeconfig.

**Comment by VioZhang** [comment]
Is it would be better that we can also support uploading the yaml of kubeconfig and the token ，not only the text？

**Comment by wenxin-01** [comment]
Issue fixed in `v3.3.0-rc.0`
